crowbar_framework/app/controllers/setups_controller.rb

```diff
@@ -22,7 +22,32 @@ def show
   end
 
   def stop
-    # TODO: place function call
+    # Stop services at the nodes
+    @service_object = CrowbarService.new logger
+
+    @service_object.shutdown_openstack_at_nodes
+
+    # Dump of the database could fail with the message that there's not enough space
+    # We need to show the error to user so he can manually prepare the space on node
+    # and than redo this step.
+    begin
+      @service_object.dump_database
+    rescue => e
+      flash[:alert] = e.message
+      # FIXME redirect to previous step?
```

Comment:
Style/CommentAnnotation: Annotation keywords like `FIXME` should be all upper case, followed by a colon, and a space, then a note describing the problem. (https://github.com/bbatsov/ruby-style-guide#annotate-keywords)